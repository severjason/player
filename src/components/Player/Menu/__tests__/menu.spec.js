// @flow
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
// test file
import { shallow } from 'enzyme';

import React from 'react';
import PlayerMenu from '../index';

type Props = {
  playlistOpened: boolean,
  userLoggedIn: boolean,
  username: string,
  showConfirmation: boolean,
  toggleConfirmation: () => boolean,
  togglePlaylist: () => void,
  logout: () => void,
};

const props: Props = {
  playlistOpened: false,
  userLoggedIn: false,
  username: 'username',
  showConfirmation: false,
  toggleConfirmation: jest.fn(),
  togglePlaylist: jest.fn(),
  logout: jest.fn(),
};

test ('renders correctly', () => {
  const wrapper = shallow(
    <PlayerMenu {...props} />
  );
  expect(wrapper).toMatchSnapshot();
});
