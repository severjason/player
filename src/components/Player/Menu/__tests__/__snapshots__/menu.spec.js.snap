// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayerMenu
    logout={[Function]}
    playlistOpened={false}
    showConfirmation={false}
    toggleConfirmation={[Function]}
    togglePlaylist={[Function]}
    userLoggedIn={false}
    username="username"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Confirmation
          confirmed={[Function]}
          showConfirmation={false}
          toggleConfirmation={[Function]}
/>,
        <div
          className="menu-item-group"
>
          <div
                    className="menu-item "
                    onClick={[Function]}
          >
                    <MdQueueMusic />
          </div>
</div>,
        <div
          className="menu-item-group"
>
          <Link
                    replace={false}
                    to="/search"
          >
                    <MdSearch />
          </Link>
</div>,
      ],
      "justifyContent": "space-between",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "confirmed": [Function],
          "showConfirmation": false,
          "toggleConfirmation": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="menu-item "
              onClick={[Function]}
>
              <MdQueueMusic />
</div>,
            null,
          ],
          "className": "menu-item-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <MdQueueMusic />,
              "className": "menu-item ",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/search"
>
            <MdSearch />
</Link>,
          "className": "menu-item-group",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <MdSearch />,
            "replace": false,
            "to": "/search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Confirmation
            confirmed={[Function]}
            showConfirmation={false}
            toggleConfirmation={[Function]}
/>,
          <div
            className="menu-item-group"
>
            <div
                        className="menu-item "
                        onClick={[Function]}
            >
                        <MdQueueMusic />
            </div>
</div>,
          <div
            className="menu-item-group"
>
            <Link
                        replace={false}
                        to="/search"
            >
                        <MdSearch />
            </Link>
</div>,
        ],
        "justifyContent": "space-between",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "confirmed": [Function],
            "showConfirmation": false,
            "toggleConfirmation": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="menu-item "
                onClick={[Function]}
>
                <MdQueueMusic />
</div>,
              null,
            ],
            "className": "menu-item-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MdQueueMusic />,
                "className": "menu-item ",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/search"
>
              <MdSearch />
</Link>,
            "className": "menu-item-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <MdSearch />,
              "replace": false,
              "to": "/search",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
